bridge/client/app/_views/ktb-sequence-view/ktb-sequence-view.component.spec.ts

```diff
@@ -31,142 +77,130 @@ describe('KtbEventsListComponent', () => {
+    /* @ts-ignore */ // Ignore private property
+    component.project = ProjectsMock[0];
+    // @ts-ignore // Ignore private property
+    component.project.sequences = SequencesMock;
+    /* eslint-enable @typescript-eslint/ban-ts-comment */
+    component._seqFilters = activeFilter;
 
-    const loadingIndicator = fixture.nativeElement.querySelector('[uitestid=keptn-loadingSequences]');
-    const emptyStateContainer = fixture.nativeElement.querySelector('[uitestid=keptn-noSequences]');
-    const sequenceList = fixture.nativeElement.querySelector('[uitestid=keptn-sequence-view-roots]');
+    // when
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore // Ignore private property
+    component.mapServiceFilters(SequenceMetadataMock);
 
-    // then
-    expect(loadingIndicator).toBeFalsy();
-    expect(emptyStateContainer).toBeTruthy();
-    expect(sequenceList).toBeFalsy();
+    expect(component._seqFilters).toEqual(activeFilter);
   });
 
-  it('should show list of sequences', async () => {
+  it('should add a service if it is in a sequence but not in metadata', () => {
     // given
-    dataService.loadProjects();
-    const projectsRequest = httpMock.expectOne('./api/controlPlane/v1/project?disableUpstreamSync=true&pageSize=50');
-    projectsRequest.flush({
-      projects: ProjectsMock,
-      totalCount: ProjectsMock.length,
-    });
-
-    const sequencesRequest = httpMock.expectOne(`./api/controlPlane/v1/sequence/${projectName}?pageSize=25`);
-    sequencesRequest.flush({
-      states: SequencesMock,
-      totalCount: SequencesMock.length,
-    });
-
-    httpMock.verify();
-    fixture.detectChanges();
+    const metadata = SequenceMetadataMock;
+    metadata.filter.services = metadata.filter.services.slice(0, 1); // remove carts-db
+    /* eslint-disable @typescript-eslint/ban-ts-comment */
+    /* @ts-ignore */ // Ignore private property
+    component.project = ProjectsMock[0];
+    // @ts-ignore // Ignore private property
+    component.project.sequences = SequencesMock;
+    /* eslint-enable @typescript-eslint/ban-ts-comment */
+    const filter = activeFilter;
+    filter[0][0].autocomplete = filter[0][0].autocomplete.slice(0, 1); // remove carts-db
+    component._seqFilters = filter;
 
-    const loadingIndicator = fixture.nativeElement.querySelector('[uitestid=keptn-loadingSequences]');
-    const emptyStateContainer = fixture.nativeElement.querySelector('[uitestid=keptn-noSequences]');
-    const sequenceList = fixture.nativeElement.querySelector('[uitestid=keptn-sequence-view-roots]');
+    // when
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore // Ignore private property
+    component.mapServiceFilters(metadata);
 
     // then
-    expect(loadingIndicator).toBeFalsy();
-    expect(emptyStateContainer).toBeFalsy();
-    expect(sequenceList).toBeTruthy();
+    expect(component._seqFilters).toEqual(activeFilter);
```

Comment:
Both are referencing the same array.
`filter[0][0].autocomplete = filter[0][0].autocomplete.slice(0, 1);` also touches the original one.
No matter how the filter will be changed, it will be as the same as `activeFilter`